base/logging_test.go

```diff
@@ -57,6 +59,71 @@ func Benchmark_LoggingPerformance(b *testing.B) {
 	}
 }
 
+func TestLogRotationInterval(t *testing.T) {
+	LongRunningTest(t)
+
+	// override min rotation interval for testing
+	originalMinRotationInterval := minLogRotationInterval
+	minLogRotationInterval = time.Millisecond * 10
+	defer func() { minLogRotationInterval = originalMinRotationInterval }()
+
+	rotationInterval := time.Millisecond * 100
+	config := &FileLoggerConfig{
+		Enabled:             BoolPtr(true),
+		CollationBufferSize: IntPtr(0),
+		Rotation: logRotationConfig{
+			RotationInterval: NewConfigDuration(rotationInterval),
+			Compress:         BoolPtr(false),
+		},
+	}
+
+	// On Windows, cleanup of t.TempDir() fails due to open log file handle from Lumberjack. Cannot be fixed from SG.
+	// https://github.com/natefinch/lumberjack/issues/185
+	var logPath string
+	if runtime.GOOS == "windows" {
+		var err error
+		logPath, err = os.MkdirTemp("", t.Name())
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			if err := os.RemoveAll(logPath); err != nil {
+				// log instead of error because it's likely this is going to fail on Windows for this test.
```

Comment:
```suggestion
				// log instead of error because it's likely this is going to fail on Windows for this test if lumberjack has not finished rotation. The `time.Sleep` at the end of the tests tries to mitigate this.
```

I wonder if the windows build machines are set up to clear temporary files, this could end up filing up the disks on the machine. We could consider writing to cwd if the builds do a fresh checkpoint of sync_gateway or do a `git clean -fdx`

Any reason here to use `os.MkdirTemp` instead of `t.TempDir` ?

Reply:
`t.TempDir` fails because it asserts there was no error from `RemoveAll`. I originally wrote it like that.

Reply:
I battled with this for a day and ended up on this solution. The sleep doesn't prevent this error, because Lumberjack keeps the current file handle open for some reason, even after calling `Close()`

On *nix this doesn't matter because the OS will still let us remove the file/directory.